languages/julia/exercises/concept/fibonacci-iterator/.docs/introduction.md

```diff
@@ -0,0 +1,102 @@
+## iterator-protocol
+
+<!-- TODO: Motivate why iterators are useful -->
+
+To implement the informal iteration interface, you need to define two methods `iterate(iter::MyIter)` and `iterate(iter::MyIter, state)`.
+Both methods must return a tuple of an item and the state of the iterator.
+The first method will return the first item and state, while the second method will return the next item and state.
+The iteration ends if `iterate` returns `nothing`.
+
+!!! note
+    Julia currently doesn't have a way to formally define interfaces.
+    To implement an interface, you need to look up which methods need to be defined in the Manual, or in the documentation of the package that "defines" the interface.
+
+!!! note
+    Julia iterators track the state decoupled from the iterator object.
+    The iterator object itself is generally not mutated.
+
+To make it clearer when the `iterate` methods are called, one can look at the translation of the `for`-loop syntax:
+
+```julia
+for item in iter
+    println(item)
+end
+```
+
+will be translated to
+
+```julia
+next_item = iterate(iter)
+
+while !isnothing(next_item)
+    (item, state) = next_item
+    
+    println(item)
+
+    next_item = iterate(iter, state)
+end
+```
+
+### Example
+
+We want to define an iterator `Squares(n)` to iterate the sequence of [square numbers](https://en.wikipedia.org/wiki/Square_number) smaller than `n`, so that:
```

Comment:
```suggestion
We want to define an iterator `Squares(n)` to iterate the sequence of [square numbers](https://en.wikipedia.org/wiki/Square_number) smaller than `n`. In mathematics, a square number is any number that is the square of an integer. For example, 9 is a square number, since it can be written as 3 × 3.  The end result should look like:
```